app/src/main/java/org/mozilla/fenix/wallpapers/WallpapersUseCases.kt

```diff
@@ -461,17 +462,19 @@ class WallpapersUseCases(
          *
          * @param wallpaper The selected wallpaper.
          */
-        override suspend fun invoke(wallpaper: Wallpaper) {
-            if (wallpaper == Wallpaper.Default || fileManager.wallpaperImagesExist(wallpaper)) {
+        override suspend fun invoke(wallpaper: Wallpaper): Wallpaper.ImageFileState {
+            return if (wallpaper == Wallpaper.Default || fileManager.wallpaperImagesExist(wallpaper)) {
                 selectWallpaper(wallpaper)
                 dispatchDownloadState(wallpaper, Wallpaper.ImageFileState.Downloaded)
+                Wallpaper.ImageFileState.Downloaded
             } else {
                 dispatchDownloadState(wallpaper, Wallpaper.ImageFileState.Downloading)
                 val result = downloader.downloadWallpaper(wallpaper)
                 dispatchDownloadState(wallpaper, result)
                 if (result == Wallpaper.ImageFileState.Downloaded) {
                     selectWallpaper(wallpaper)
                 }
+                result
```

Comment:
I think it would be worth updating the tests to capture the return behavior here.